### PR TITLE
aws - cost optimization filter

### PR DESCRIPTION
**Pull request by kapilt**
initial focus on resource level optimizations, versus ri/savings plan on account.

### CODE REVIEW

**Review by ajkerrigan — CHANGES_REQUESTED** [review]
LGTM aside from one point inline about handling filter-defined action types. This looks handy to support, thanks! :+1:

File: c7n/filters/costhub.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    permissions = ('cost-optimization-hub:ListRecommendations',)
+    annotation = "c7n:cost_optimize"
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('cost-optimization-hub')
+        id_field = self.manager.resource_type.id
+        filter_params = filter_empty({
+            'actionTypes': (
+                self.manager.data.get('action') and [self.manager.data.get('action')]
```
[the remark this answers is not in the document]
Comment (kapilt) [reply c7n/filters/costhub.py:82]: to simplify remediation policies, I'm going to force this to a scalar

File: c7n/filters/costhub.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    """
+
+    schema = type_schema(
+        'cost-optimization',
+        efforts={
+            'type': 'array',
+            'items': {'enum': ['VeryLow', 'Low', 'Medium', 'High', 'VeryHigh']},
+        },
+        action={
```
[the remark this answers is not in the document]
Comment (kapilt) [reply c7n/filters/costhub.py:44]: with a move to a scalar action extant name should be okay.

**Review by ajkerrigan — APPROVED** [review]
Cool cool, looks good to me :+1:

**Pull request merged**